src/hex/coordinate.rs

```diff
@@ -29,6 +35,46 @@ pub enum HexDirection {
     FrontLeft,
 }
 
+impl From<i32> for HexDirection {
+    fn from(value: i32) -> Self {
+        match value % 6 {
+            -6 | 0 | 6 => HexDirection::Front,
```

Comment:
What is the case where value % 6 could be -6?

Reply:
Fixed with the refactor into `value.rem_euclid(6)`